services/node_manager/src/main/java/com/futurewei/alcor/nodemanager/entity/NodeInfo.java

```diff
@@ -0,0 +1,99 @@
+/*Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.nodemanager.entity;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.futurewei.alcor.nodemanager.utils.NodeManagerConstant;
+import lombok.Data;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Data
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY, getterVisibility = JsonAutoDetect.Visibility.NONE,
+        setterVisibility = JsonAutoDetect.Visibility.NONE, creatorVisibility = JsonAutoDetect.Visibility.NONE)
+public class NodeInfo implements Serializable {
+    private static final Logger logger = LoggerFactory.getLogger(NodeInfo.class);
+
+    @JsonProperty("node_id")
+    private String id;
+
+    @JsonProperty("node_name")
+    private String name;
+
+    @JsonProperty("local_ip")
+    private InetAddress localIp;
```

Comment:
Any reason that we want to pass InetAddress as an parameter? 

The object is quite complicated. 

{
  "host_info": {
    "local_ip": {
      "address": [
        "string"
      ],
      "anyLocalAddress": true,
      "canonicalHostName": "string",
      "hostAddress": "string",
      "hostName": "string",
      "linkLocalAddress": true,
      "loopbackAddress": true,
      "mcglobal": true,
      "mclinkLocal": true,
      "mcnodeLocal": true,
      "mcorgLocal": true,
      "mcsiteLocal": true,
      "multicastAddress": true,
      "siteLocalAddress": true
    },
    "mac_address": "string",
    "node_id": "string",
    "node_name": "string",
    "server_port": 0,
    "veth": "string"
  }
}

Reply:
IP address type is changed to String